plugins/lookup/passwordstore.py

```diff
@@ -328,8 +367,26 @@ def get_passresult(self):
             else:
                 return None
 
+    @contextmanager
+    def opt_lock(self, type):
+        if self.get_option('lock') == type:
+            tmpdir = os.environ.get('TMPDIR', '/tmp')
```

Comment:
Doing the conversion inside `opt_lock` is really not a good idea. Such things should really be done on setup time. If the user for example sets `lock=write` and an invalid value for `locktimeout`, the plugin already performed a read operation before finally telling the user that `locktimeout` is bad, or might not tell the user at all because no write was needed.

Reply:
I’m aware of that,  but I didn’t consider it to be problem - after all, read operations aren’t destructive and if locking isn’t required, the timeout doesn’t matter either.

Reply:
Refactored once again